### DISCUSSION TITLE
Extractors

### DISCUSSION THREAD

**Discussion by longfellowone** [async-graphql/async-graphql#837] [Ideas]
Would it make sense to use the extractor pattern to pull .data() from &Context<'_>

Not sure if its even possible?

Similar to the way actix and axum work

https://docs.rs/axum/latest/axum/extract/index.html
https://actix.rs/docs/extractors/

## Code example

Current method

```rs
#[Object]
impl UserQueries {
    pub async fn me(&self, ctx: &Context<'_>) -> async_graphql::Result<User> {
        let user = ctx.data_unchecked::<User>();
        // ...
    }
}

```

With extractor (Instead of Data could be name anything, like `FromCtx<>` or `Ctx<>`

```rs
#[Object]
impl UserQueries {
    pub async fn me(&self, user: Data<User> ) -> async_graphql::Result<User> {
      //  ...
    }
}

```

**Comment by Sytten** [comment]
I think this is the default pattern these days and it makes a lot of sense. @sunli829 any idea if that would be hard to implement?
